cloud/shared/bin/lib/log.sh

```diff
@@ -0,0 +1,97 @@
+#! /usr/bin/env bash
+
+#######################################
+# Record a deployment success event in the log.
+# Args:
+#   1: The tag of the deployed server binary.
+#   2: Identifier for the user who initiated the command.
+# Globals read:
+#   LOG_TEMPFILE
+#######################################
+function log::deploy_succeeded() {
+  log::ensure_log_file_fetched
+  log::check_deploy_args "$@"
+
+  echo "$(log::timestamp) DEPLOY SUCCESS ${1} ${2}" >> "${LOG_TEMPFILE}"
+}
+
+#######################################
+# Record a deployment failure event in the log.
+# Args:
+#   1: The tag of the deployed server binary.
+#   2: Identifier for the user who initiated the command.
+# Globals read:
+#   LOG_TEMPFILE
+#######################################
+function log::deploy_failed() {
+  log::ensure_log_file_fetched
+  log::check_deploy_args "$@"
+
+  echo "$(log::timestamp) DEPLOY FAILED ${1} ${2}" >> "${LOG_TEMPFILE}"
+}
+
+#######################################
+# Record the initialization event in the log.
+# Args:
+#   1: Identifier for the user who initiated the command.
+# Globals read:
+#   LOG_TEMPFILE
+#######################################
+function log::initialized() {
+  log::ensure_log_file_fetched
+  log::validate_token "${1}"
+
+  echo "$(log::timestamp) INITIALIZED SUCCESS ${1}" >> "${LOG_TEMPFILE}"
+}
+
+#######################################
+# Check that at least two arguments exist and do not contain spaces.
```

Comment:
question (blocking): should we check arguments after the 2nd one (if there are any) to make sure they don't contain spaces? Right now it looks like it's only checking the first two.

Reply:
Done